src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,11 +1,23 @@
 package core.basesyntax.service;
 
+import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private final StorageDaoImpl storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
+        if (user == null
+                || user.getLogin() == null
+                || user.getPassword() == null) {
+            throw new RuntimeException("Invalid data");
+        }
+        if (storageDao.get(user.getLogin()) == null
+                && user.getPassword().length() >= 6
+                && user.getAge() >= 18) {
```

Comment:
In case of invalid data you should throw RuntimeException.

So you need to throw it also at second if in case they are not allowed.

It seems to be better to use the TDD principle in this task.